nimble/controller/src/ble_ll_sync.c

```diff
@@ -117,6 +122,10 @@ struct ble_ll_sync_sm {
     uint16_t event_cntr_last_received;
     uint8_t adv_addr_rpa[6];
 #endif
+
+#if MYNEWT_VAL(BLE_LL_CFG_FEAT_LL_BIGINFO_REPORTS)
+    struct ble_ll_acad_biginfo biginfo;
+#endif
```

Comment:
there's no need to keep this huge structure in sm all the time. all you need is likely to check if biginfo is present in acad, store the offset and then parse it directly to hci event.

for now I think it will be enough to return that offset from `ble_ll_sync_check_acad` and use it in `ble_ll_sync_rx_pkt_in` to check whether to send biginfo report

Reply:
no need to keep offset in sm, `ble_ll_sync_check_acad` is called from `ble_ll_sync_rx_pkt_in` so it can just return pointer to biginfo and biginfo length